mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -852,7 +852,9 @@ void CMomentumPlayer::CheckChatText(char *p, int bufsize) { g_pMomentumGhostClie
 // Overrides Teleport() so we can take care of the trail
 void CMomentumPlayer::Teleport(const Vector *newPosition, const QAngle *newAngles, const Vector *newVelocity)
 {
-    if (!m_bAllowUserTeleports || (m_Data.m_bMapFinished && !m_cvarMapFinMoveEnable.GetBool()))
+    int nButtons = m_nButtons;
+    if (!m_bAllowUserTeleports || (m_Data.m_bMapFinished && !m_cvarMapFinMoveEnable.GetBool()) ||
+        (m_bHasPracticeMode && (nButtons & IN_JUMP)))
```

Comment:
Let's break these up into separate bools and early return for each, so this code is way easier to read

Reply:
And no need to use a separate int for the buttons